lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -59,15 +60,16 @@ private MapTaskHost(IMapFunction<TMapInput, TMapOutput> mapTask, IGroupCommClien
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            MapInputWithControlMessage<TMapInput> mapInput = _dataAndMessageReceiver.Receive();
-
-            while (mapInput.ControlMessage == MapControlMessage.AnotherRound)
+            while (true)
             {
+                GC.Collect();
```

Comment:
lemme do the measurements

On Mon, Sep 28, 2015 at 3:25 PM, Anupam <notifications@github.com> wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs
> <https://github.com/apache/incubator-reef/pull/527#discussion_r40613652>:
>
> >              {
> > +                GC.Collect();
>
> When to call GC.Collect
> <http://blogs.msdn.com/b/ricom/archive/2004/11/29/271829.aspx>
> What is wrong about using GC.Collect
> <http://stackoverflow.com/a/118776/285199>
> Best practice for forcing Garbage Collection in C#
> <http://stackoverflow.com/a/233647/285199>
>
> This might be a correct usage of GC.Collect but you are not citing any
> measurement that you have done to convince us that it indeed is. Without
> that measurement I don't think this change makes sense.
>
> If you read the first link on "When to call GC.Collect", Rule#1 trumps
> Rule#2 when you have not given enough evidence. Further, rule#2 makes
> assumptions about a program's runtime characteristics which are very hard
> to codify or enforce, hence the usage is brittle with very hard to measure
> side effects.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/527/files#r40613652>.
>